pkg/gatewayserver/io/basicstationlns/messages/util.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package messages
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	"go.thethings.network/lorawan-stack/pkg/band"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+var errDataRateIndex = errors.Define("data_rate_index", "datarate index is out of range")
+
+func getInt32IntegerAsByteSlice(value int32) ([]byte, error) {
+	b := new(bytes.Buffer)
+	err := binary.Write(b, binary.LittleEndian, value)
+	if err != nil {
+		return nil, err
+	}
+	return b.Bytes(), nil
+}
+
+func getDataRateFromDataRateIndex(bandID string, index int) (ttnpb.DataRate, error) {
```

Comment:
`getDataRateFromIndex` is descriptive enough

Reply:
Let's follow the same naming scheme as NS already does? https://github.com/TheThingsNetwork/lorawan-stack/blob/master/pkg/networkserver/utils.go#L39-L59